### PR TITLE
Added in my citation check code

### PR DESCRIPTION
**Pull request by andrewholding**
Added in my code for the Citation check. It currently is very basic, but forms the tools needed.

### CODE REVIEW

File: 4-citation/check_citation.py [hunk 1]
```diff
@@ -1,16 +1,54 @@
...
+def YAML_check(filename):
+    with open(filename) as file:
+	try:
+		citation = yaml.safe_load(file)
+		return "YAML file loaded"
+	except yaml.YAMLError as exc:
+        	print(exc)
+		return "YAML failed to load"
+
+define Contents_check(filename):
```
Comment (c-martinez) [inline 4-citation/check_citation.py:25]: Shouldn't this be `def` ?

File: 4-citation/check_citation.py [hunk 2]
```diff
@@ -1,16 +1,54 @@
...
+		print("Passed")
+		return "Citation Scheme passed"	
+	except	 MultipleInvalid as e:
+		exc = e
+		print ("Error: " + str(exc))
+		return "Citation Schema not Valid"
 
 
+def main():
+    print('Citation Check)
```
Comment (c-martinez) [inline 4-citation/check_citation.py:47]: Missing `'`

File: 4-citation/check_citation.py [hunk 3]
```diff
@@ -1,16 +1,54 @@
...
+    print(filename = 'CITATION.cff')
```
Comment (c-martinez) [inline 4-citation/check_citation.py:48]: I'm guessing you need to do something like:
```
filename = 'CITATION.cff'
print(filename)
```

File: 4-citation/check_citation.py [hunk 4]
```diff
@@ -1,16 +1,54 @@
...
+        	print(exc)
+		return "YAML failed to load"
+
+define Contents_check(filename):
+	citation = yaml.safe_load(file)
+	s = Schema({
+		Required("doi"): str,
+		Required("title"): str,
+		Optional("version"): str,
+		Required("authors"): str,
```
Comment (c-martinez) [inline 4-citation/check_citation.py:31]: There will be an issue here, because `authors` is not a string but a dictionary.

**Review by c-martinez — CHANGES_REQUESTED** [review]
Left some comments.

**Comment by andrewholding** [comment]
Fixed and merged in newer fork.

**Pull request closed without merging**